test/new-e2e/tests/windows/service-test/startstop_test.go

```diff
@@ -0,0 +1,335 @@
+	// check the Application event log for timeout messages
+	out, err = windowsCommon.GetEventLogErrorsAndWarnings(host, "Application")
+	s.Require().NoError(err, "should get errors and warnings from Application event log")
+	s.T().Logf("Errors and warnings from Application event log:\n%s", out)
+	s.Assert().NotContains(out, "hard stopping service", "should not have timeout messages in the event log")
+}
+
+// TestAgentStartsAllServices tests that starting the agent starts all services
+func (s *startStopTestSuite) TestAgentStartsAllServices() {
+	s.startAgent()
+	s.requireAllServicesState("Running")
+}
+
+// TestAgentStopsAllServices tests that stopping the agent stops all services
+func (s *startStopTestSuite) TestAgentStopsAllServices() {
+	host := s.Env().RemoteHost
+	s.startAgent()
+	s.requireAllServicesState("Running")
+
+	// stop the agent
+	err := windowsCommon.StopService(host, "datadogagent")
+	s.Require().NoError(err, "should stop the datadogagent service")
+
+	// ensure all services are stopped
+	s.assertAllServicesState("Stopped")
+}
+
+// TestHardExitEventLogEntry tests that the System event log contains an "unexpectedly terminated" message when a service is killed
+func (s *startStopTestSuite) TestHardExitEventLogEntry() {
+	host := s.Env().RemoteHost
+	s.startAgent()
+	s.requireAllServicesState("Running")
+
+	// kill the agent
+	for _, serviceName := range s.expectedUserServices() {
+		// get pid
+		pid, err := windowsCommon.GetServicePID(host, serviceName)
+		s.Require().NoError(err, "should get the PID for %s", serviceName)
+		// kill the process
+		_, err = host.Execute(fmt.Sprintf("Stop-Process -Force -Id %d", pid))
+		s.Require().NoError(err, "should kill the process with PID %d", pid)
+		// service should stop
+		status, err := windowsCommon.GetServiceStatus(host, serviceName)
+		s.Require().NoError(err, "should get the status for %s", serviceName)
+		s.Require().Equal("Stopped", status, "%s should be stopped", serviceName)
+	}
+
+	// collect display names for services
+	displayNames := make([]string, 0, len(s.expectedUserServices()))
+	for _, serviceName := range s.expectedUserServices() {
+		conf, err := windowsCommon.GetServiceConfig(host, serviceName)
+		s.Require().NoError(err, "should get the configuration for %s", serviceName)
+		displayNames = append(displayNames, conf.DisplayName)
+	}
+
+	// check the System event log for hard exit messages
+	s.Assert().EventuallyWithT(func(c *assert.CollectT) {
+		out, err := windowsCommon.GetEventLogErrorsAndWarnings(host, "System")
+		require.NoError(c, err, "should get errors and warnings from System event log")
+		s.T().Logf("Errors and warnings from System event log:\n%s", out)
+		for _, displayName := range displayNames {
+			match := fmt.Sprintf("The %s service terminated unexpectedly", displayName)
+			assert.Contains(c, out, match, "should have hard exit messages in the event log")
+		}
+	}, 1*time.Minute, 1*time.Second, "should have hard exit messages in the event log")
+}
```

Comment:
I wonder about utility of this since you are testing SCM behavior and not our own, right?

Reply:
That's right, I was wondering the same.

I leant toward keeping it as a way to verify that we're able to find the event log messages. Since `TestStopTimeout` passes on the absence of those messages I'm a little concerned about that check breaking and the test "passing".

I manually tested the failure case by adding a delay to the agent shutdown, but we can't test that in an ongoing fashion, unless we add an env var to tell the agent to hang on shutdown, which feels weird.

What do you think?

Reply:
It is fine, I was curious